common/src/x509.rs

```diff
@@ -0,0 +1,79 @@
+// Copyright 2021 Oxide Computer Company
+use std::fs::File;
+use std::io::{self, BufReader};
+
```

Comment:
I don't know enough about TLS to know if we are doing this correctly.  This is probably fine for our initial prototype, but maybe we want to get Rick or someone who knows this stuff better to verify it?

In the longer term, I wonder if we will have some common library that any Oxide service can use to leverage TLS?  Do we think that might happen?

Reply:
I'm hoping that `with_safe_defaults` delegates some of that responsibility to rustls' maintainers, but I agree that this should be verified. I know enough about TLS to probably be dangerous but can't claim expertise.

Reply:
let's file an issue for ourselves so we don't forget about this.

Reply:
Opened https://github.com/oxidecomputer/crucible/issues/187